tests fail on python 3.8

```
==> Starting check()...
============================= test session starts ==============================
platform linux -- Python 3.8.0, pytest-5.2.2, py-1.8.0, pluggy-0.13.0
rootdir: /build/python-dephell-discover/src/dephell_discover-0.2.8
collected 27 items

tests/test_metainfo.py FFFFF.                                            [ 22%]
tests/test_root.py .....................                                 [100%]

=================================== FAILURES ===================================
__________________ test_vars[__license__ = "MIT"-license-MIT] __________________

tmp_path = PosixPath('/tmp/pytest-of-builduser/pytest-0/test_vars___license______MIT__0')
content = '__license__ = "MIT"', name = 'license', expected = 'MIT'

    @pytest.mark.parametrize('content, name, expected', [
        ('__license__ = "MIT"', 'license', 'MIT'),
    
        ('__version__ = "0.2.7"', 'version', '0.2.7'),
        ('__version__ = ("0", "2", "7")', 'version', '0.2.7'),
        # ('__version__ = (0, 2, 7)', 'version', '0.2.7'),
    
        ('__author__ = "Gram"', 'authors', ['Gram']),
        ('__authors__ = ["Gram"]', 'authors', ['Gram']),
    
    ])
    def test_vars(tmp_path: Path, content: str, name: str, expected):
        path = tmp_path / 'lol.py'
        path.write_text(content)
        info = MetaInfo.parse(paths=[path])
>       assert getattr(info, name) == expected
E       AssertionError: assert None == 'MIT'
E        +  where None = getattr(MetaInfo(lines=[]), 'license')

tests/test_metainfo.py:23: AssertionError
________________ test_vars[__version__ = "0.2.7"-version-0.2.7] ________________

tmp_path = PosixPath('/tmp/pytest-of-builduser/pytest-0/test_vars___version______0_2_70')
content = '__version__ = "0.2.7"', name = 'version', expected = '0.2.7'

    @pytest.mark.parametrize('content, name, expected', [
        ('__license__ = "MIT"', 'license', 'MIT'),
    
        ('__version__ = "0.2.7"', 'version', '0.2.7'),
        ('__version__ = ("0", "2", "7")', 'version', '0.2.7'),
        # ('__version__ = (0, 2, 7)', 'version', '0.2.7'),
    
        ('__author__ = "Gram"', 'authors', ['Gram']),
        ('__authors__ = ["Gram"]', 'authors', ['Gram']),
    
    ])
    def test_vars(tmp_path: Path, content: str, name: str, expected):
        path = tmp_path / 'lol.py'
        path.write_text(content)
        info = MetaInfo.parse(paths=[path])
>       assert getattr(info, name) == expected
E       AssertionError: assert None == '0.2.7'
E        +  where None = getattr(MetaInfo(lines=[]), 'version')

tests/test_metainfo.py:23: AssertionError
____________ test_vars[__version__ = ("0", "2", "7")-version-0.2.7] ____________

tmp_path = PosixPath('/tmp/pytest-of-builduser/pytest-0/test_vars___version_______0___0')
content = '__version__ = ("0", "2", "7")', name = 'version', expected = '0.2.7'

    @pytest.mark.parametrize('content, name, expected', [
        ('__license__ = "MIT"', 'license', 'MIT'),
    
        ('__version__ = "0.2.7"', 'version', '0.2.7'),
        ('__version__ = ("0", "2", "7")', 'version', '0.2.7'),
        # ('__version__ = (0, 2, 7)', 'version', '0.2.7'),
    
        ('__author__ = "Gram"', 'authors', ['Gram']),
        ('__authors__ = ["Gram"]', 'authors', ['Gram']),
    
    ])
    def test_vars(tmp_path: Path, content: str, name: str, expected):
        path = tmp_path / 'lol.py'
        path.write_text(content)
        info = MetaInfo.parse(paths=[path])
>       assert getattr(info, name) == expected
E       AssertionError: assert None == '0.2.7'
E        +  where None = getattr(MetaInfo(lines=[]), 'version')

tests/test_metainfo.py:23: AssertionError
_______________ test_vars[__author__ = "Gram"-authors-expected3] _______________

tmp_path = PosixPath('/tmp/pytest-of-builduser/pytest-0/test_vars___author______Gram__0')
content = '__author__ = "Gram"', name = 'authors', expected = ['Gram']

    @pytest.mark.parametrize('content, name, expected', [
        ('__license__ = "MIT"', 'license', 'MIT'),
    
        ('__version__ = "0.2.7"', 'version', '0.2.7'),
        ('__version__ = ("0", "2", "7")', 'version', '0.2.7'),
        # ('__version__ = (0, 2, 7)', 'version', '0.2.7'),
    
        ('__author__ = "Gram"', 'authors', ['Gram']),
        ('__authors__ = ["Gram"]', 'authors', ['Gram']),
    
    ])
    def test_vars(tmp_path: Path, content: str, name: str, expected):
        path = tmp_path / 'lol.py'
        path.write_text(content)
        info = MetaInfo.parse(paths=[path])
>       assert getattr(info, name) == expected
E       AssertionError: assert [] == ['Gram']
E         Right contains one more item: 'Gram'
E         Use -v to get the full diff

tests/test_metainfo.py:23: AssertionError
_____________ test_vars[__authors__ = ["Gram"]-authors-expected4] ______________

tmp_path = PosixPath('/tmp/pytest-of-builduser/pytest-0/test_vars___authors_______Gram0')
content = '__authors__ = ["Gram"]', name = 'authors', expected = ['Gram']

    @pytest.mark.parametrize('content, name, expected', [
        ('__license__ = "MIT"', 'license', 'MIT'),
    
        ('__version__ = "0.2.7"', 'version', '0.2.7'),
        ('__version__ = ("0", "2", "7")', 'version', '0.2.7'),
        # ('__version__ = (0, 2, 7)', 'version', '0.2.7'),
    
        ('__author__ = "Gram"', 'authors', ['Gram']),
        ('__authors__ = ["Gram"]', 'authors', ['Gram']),
    
    ])
    def test_vars(tmp_path: Path, content: str, name: str, expected):
        path = tmp_path / 'lol.py'
        path.write_text(content)
        info = MetaInfo.parse(paths=[path])
>       assert getattr(info, name) == expected
E       AssertionError: assert [] == ['Gram']
E         Right contains one more item: 'Gram'
E         Use -v to get the full diff

tests/test_metainfo.py:23: AssertionError
========================= 5 failed, 22 passed in 1.95s =========================

```


Broken due to use of ast.Str, which doesn't match type checks any more due to https://bugs.python.org/issue32892

Used here:

https://github.com/dephell/dephell_discover/blob/95eaa395f1cc23e38fe36dbde94d5b4b32992766/dephell_discover/_line.py#L56
https://github.com/dephell/dephell_discover/blob/95eaa395f1cc23e38fe36dbde94d5b4b32992766/dephell_discover/_line.py#L68

P.S. Why does this use `type(foo) is bar` instead of `isinstance(foo, bar)`?

--- a/.travis.yml
+++ b/.travis.yml
@@ -23,6 +23,8 @@ matrix:
       env: ENV=pytest
     - python: "3.7-dev"
       env: ENV=pytest
+    - python: "3.8-dev"
+      env: ENV=pytest
     - python: "pypy3.5"
       env: ENV=pytest
 

--- a/dephell_discover/_line.py
+++ b/dephell_discover/_line.py
@@ -7,10 +7,15 @@ import attr
 
 from ._constants import DOCSTRING
 
-if sys.version_info[:2] >= (3,8):
-    _ast_str = ast.Constant
-else:
-    _ast_str = ast.Str
+
+def _get_str(node) -> Optional[str]:
+    if type(node) is ast.Str:
+        return node.s
+    if sys.version_info[:2] >= (3, 8):
+        if type(node) is getattr(ast.Constant):
+            return node.value
+    return None
+
 
 @attr.s(frozen=True)
 class Line:
@@ -55,10 +60,11 @@ class Line:
         if target.id not in cls._vars:
             return None
 
-        value = tree.body[0].value  # type: ignore
+        expr = tree.body[0].value  # type: ignore
 
         # get string
-        if type(value) is _ast_str:
+        value = _get_str(expr)
+        if value:
             return cls(
                 target=target.id,
                 value=value.s,
@@ -68,13 +74,14 @@ class Line:
             )
 
         # get list
-        if type(value) is ast.List or type(value) is ast.Tuple:
-            for element in value.elts:
-                if type(element) is not _ast_str:
+        if type(expr) is ast.List or type(expr) is ast.Tuple:
+            elements = [_get_str(element) for element in expr.elts]
+            for element in elements:
+                if element is None:
                     return None
             return cls(
                 target=target.id,
-                value=[element.s for element in value.elts],
+                value=elements,
                 content=content,
                 row=row,
                 path=path,

--- a/dephell_discover/_line.py
+++ b/dephell_discover/_line.py
@@ -9,10 +9,15 @@ from ._constants import DOCSTRING
 
 
 def _get_str(node) -> Optional[str]:
-    if type(node) is ast.Str:
-        return node.s
-    if sys.version_info[:2] >= (3, 8):
-        if type(node) is getattr(ast.Constant):
+    # python <3.8
+    if sys.version_info[:2] < (3, 8):
+        if type(node) is ast.Str:
+            return node.s
+        return None
+
+    # python >=3.8
+    if type(node) is ast.Constant:
+        if type(node.value) is str:
             return node.value
     return None
 

--- a/dephell_discover/_line.py
+++ b/dephell_discover/_line.py
@@ -1,11 +1,16 @@
 import ast
 from pathlib import Path
+import sys
 from typing import Optional
 
 import attr
 
 from ._constants import DOCSTRING
 
+if sys.version_info[:2] >= (3,8):
+    _ast_str = ast.Constant
+else:
+    _ast_str = ast.Str
 
 @attr.s(frozen=True)
 class Line:
@@ -53,7 +58,7 @@ class Line:
         value = tree.body[0].value  # type: ignore
 
         # get string
-        if type(value) is ast.Str:
+        if type(value) is _ast_str:
             return cls(
                 target=target.id,
                 value=value.s,
@@ -65,7 +70,7 @@ class Line:
         # get list
         if type(value) is ast.List or type(value) is ast.Tuple:
             for element in value.elts:
-                if type(element) is not ast.Str:
+                if type(element) is not _ast_str:
                     return None
             return cls(
                 target=target.id,

--- a/dephell_discover/_line.py
+++ b/dephell_discover/_line.py
@@ -67,7 +67,7 @@ class Line:
         if value:
             return cls(
                 target=target.id,
-                value=value.s,
+                value=value,
                 content=content,
                 row=row,
                 path=path,

